File: Rust Projects/project2/src/sorting/shellsort.rs
#[allow(dead_code)]
pub fn shellsort<T>(arr: &mut [T]) -> ()
where
    T: PartialOrd<T> + Copy,
{
    let mut gap: usize = arr.len() / 2;

    while gap > 0 {
        for i in gap..arr.len() {
            let mut j = i;

            let temp = arr[i];

            while j >= gap && arr[j - gap] > temp {
                arr[j] = arr[j - gap];

                j -= gap;
            }

            arr[j] = temp;
        }

        gap /= 2;
    }
}

File: Oriented-Object Programming/Contador/src/projeto_bimestral/Shellsort.java

package projeto_bimestral;

public class Shellsort {

    void shellsort(int[] x){
        
        for(int gap = x.length/2; gap > 0; gap /= 2){
            
            for(int i = gap; i < x.length; i += 1){
                int temp = x[i];
                
                int j;
                for(j = i; j >= gap && x[j-gap] > temp; j -= gap){
                    x[j] = x[j-gap];
                }
                
                x[j] = temp;
            }
        }
    }
    
    public static void shell_sorted(int[] x){ new Shellsort().shellsort(x);}
    
}
